gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/dsl/ProguardSettings.kt

```diff
@@ -22,4 +22,5 @@ abstract class ProguardSettings @Inject constructor(
     val configurationFiles: ConfigurableFileCollection = objects.fileCollection()
     val isEnabled: Property<Boolean> = objects.notNullProperty(false)
     val obfuscate: Property<Boolean> = objects.notNullProperty(false)
+    val optimize: Property<Boolean> = objects.notNullProperty(false)
```

Comment:
`release` builds usually should be optimized.

Can we:
- exclude only problematic functions/classes
- write test that checks all classes, to find problematic (maybe not now, but in the future)

I wouldn't make this change only to speed up release and simplify the release process.

As an example, [Android builds](https://android.googlesource.com/platform/sdk/+/master/files/proguard-android-optimize.txt) enable optimizations in the release build, and there is the rule that each third-party library should contain a valid ProGuard rules.

Reply:
But there is no such rule for desktop. With lots of external libraries it is often a PITA to get all the ProGuard rules right just in order to be able to do a release build. Maybe one should not carry over all Android habbits to the rest of the world.

Reply:
This is indeed a dillemma. I have to admit that I do not have an Android background but I have been using JavaFX with GraalVM/native-image which conceptually has a similar problem and so I am wondering whether you ever considered something like the native-image agent (https://www.graalvm.org/22.0/reference-manual/native-image/Agent/) which I found to be very helpful and easier to use than guessing ProGuard rules for libraries which are not your own. Would some kind of a bridge between the agents output and ProGuard be possible? Just thinking.

Reply:
`native-image` looks interesting

Reply:
> Can we:
> * exclude only problematic functions/classes
> * write test that checks all classes, to find problematic (maybe not now, but in the future)


In both cases most likely no.
I don't see any way to exclude just one class from optimization.
Also, simply forcing classloading does not trigger the verification error, which is described in the issue.
Also, I'm really not sure, if PG optimizations are actually beneficial with Hotspot. The desktop JVM is much better at optimizing code than mobile. Our primary motivation for supporting PG was to enable minification, not making bytecode faster

Reply:
After having  a closer look at https://www.guardsquare.com/manual/configuration/optimizations I am a bit shocked to realize what PG is actually trying to do with my code by default. This really looks like fishing for trouble to me where all I really want is a reliable build combined with some reasonable code reduction. Packaging your app with the most recent JVM is probably more beneficial and safer than applying all this bit fiddling.